### PR TITLE
Allow n-dim data in LearningWithNoisyLabels

### PR DESCRIPTION
**Pull request by anishathalye**
LearningWithNoisyLabels used `sklearn.utils.check_X_y` to enforce that `X` was 2D, to be in line with what sklearn's standard estimators expect. However, LearningWithNoisyLabels is dataset-agnostic: it doesn't look at the data points themselves. If the underlying classifier supports data in a different shape, there's no reason LearningWithNoisyLabels should prohibit it. Users have requested that we relax this unnecessary restriction [[1]] so LearningWithNoisyLabels will more naturally support e.g. image datasets and CNN models.

Thanks to @kothari1997narayan for suggesting this change.

[1]: https://github.com/cleanlab/cleanlab/issues/86

### CODE REVIEW

**Review by jwmueller — APPROVED** [review]
LGTM!

**Comment by jwmueller** [comment]
black complained though: https://github.com/cleanlab/cleanlab/runs/5730967872?check_suite_focus=true
